### PR TITLE
Use UserId from identity database

### PR DESCRIPTION
**Pull request by leonidbogin**

### CODE REVIEW

File: DIMS-Core/DIMS-Core/Controllers/TaskTrackController.cs [hunk 1]
```diff
@@ -15,25 +16,37 @@ public class TaskTrackController : Controller
...
         public async Task<IActionResult> Index()
         {
-            // To Do - Get the id of the current user
-            int userId = 3;
+            if (!User.Identity.IsAuthenticated)
+            {
+                return RedirectToAction("Index", "Home", new { });
+            }
+
+            var currentUser = memberService.GetMemberByEmail(User.Identity.Name).Result;
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core/Controllers/TaskTrackController.cs:42]: await

File: DIMS-Core/DIMS-Core/Controllers/TaskTrackController.cs [hunk 2]
```diff
@@ -57,16 +70,25 @@ public IActionResult Details(int id, string back = null)
...
         public async Task<IActionResult> Create(int userTaskId = 0, string back = null)
         {
-            // To Do - Get the id of the current user
-            int userId = 3;
+            if (!User.Identity.IsAuthenticated)
+            {
+                return RedirectToAction("Index", "Home", new { });
+            }
+
+            var currentUser = memberService.GetMemberByEmail(User.Identity.Name).Result;
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core/Controllers/TaskTrackController.cs:78]: await

File: DIMS-Core/DIMS-Core/Controllers/TaskTrackController.cs [hunk 3]
```diff
@@ -96,13 +118,22 @@ public async Task<IActionResult> Edit(int id, string back = null)
...
             }
 
-            // To Do - Get the id of the current user
-            int userId = 3;
+            if (!User.Identity.IsAuthenticated)
+            {
+                return RedirectToAction("Index", "Home", new { });
+            }
+
+            var currentUser = memberService.GetMemberByEmail(User.Identity.Name).Result;
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core/Controllers/TaskTrackController.cs:126]: await

File: DIMS-Core/DIMS-Core/Controllers/TaskTrackController.cs [hunk 4]
```diff
@@ -112,19 +143,28 @@ public async Task<IActionResult> Edit(int id, string back = null)
...
         public async Task<IActionResult> Edit([FromForm] TaskTrackViewModel model)
         {
-            // To Do - Get the id of the current user
-            int userId = 3;
+            if (!User.Identity.IsAuthenticated)
+            {
+                return RedirectToAction("Index", "Home", new { });
+            }
+
+            var currentUser = memberService.GetMemberByEmail(User.Identity.Name).Result;
```
Comment (PushMeNow) [inline DIMS-Core/DIMS-Core/Controllers/TaskTrackController.cs:151]: await

**Review by PushMeNow — CHANGES_REQUESTED** [review]
Hi.
I did some comments. Look them through and fix.

File: DIMS-Core/DIMS-Core/Controllers/TaskTrackController.cs (hunk 1 above)
Comment (leonidbogin) [reply DIMS-Core/DIMS-Core/Controllers/TaskTrackController.cs:42]:
```suggestion
            var currentUser = await memberService.GetMemberByEmail(User.Identity.Name);

File: DIMS-Core/DIMS-Core/Controllers/TaskTrackController.cs (hunk 3 above)
Comment (leonidbogin) [reply DIMS-Core/DIMS-Core/Controllers/TaskTrackController.cs:126]:
```suggestion
            var currentUser = await memberService.GetMemberByEmail(User.Identity.Name);

File: DIMS-Core/DIMS-Core/Controllers/TaskTrackController.cs (hunk 4 above)
Comment (leonidbogin) [reply DIMS-Core/DIMS-Core/Controllers/TaskTrackController.cs:151]:
```suggestion
            var currentUser = await memberService.GetMemberByEmail(User.Identity.Name);

File: DIMS-Core/DIMS-Core/Controllers/TaskTrackController.cs (hunk 2 above)
Comment (leonidbogin) [reply DIMS-Core/DIMS-Core/Controllers/TaskTrackController.cs:78]:
```suggestion
            var currentUser = await memberService.GetMemberByEmail(User.Identity.Name);

File: DIMS-Core/DIMS-Core/Controllers/TaskTrackController.cs [hunk 5]
```diff
@@ -57,16 +70,25 @@ public IActionResult Details(int id, string back = null)
         [HttpGet("create")]
         public async Task<IActionResult> Create(int userTaskId = 0, string back = null)
         {
-            // To Do - Get the id of the current user
-            int userId = 3;
+            if (!User.Identity.IsAuthenticated)
+            {
+                return RedirectToAction("Index", "Home", new { });
```
Comment (maximsan) [inline DIMS-Core/DIMS-Core/Controllers/TaskTrackController.cs:75]: Actually it's better to create `AuthAttribute` or `AuthMiddleware` to handle all auth related logic there before controller.. Please, discuss it with @PushMeNow

File: DIMS-Core/DIMS-Core/Controllers/TaskTrackController.cs [hunk 6]
```diff
@@ -112,19 +143,28 @@ public async Task<IActionResult> Edit(int id, string back = null)
...
+            }
+
+            var currentUser = await memberService.GetMemberByEmail(User.Identity.Name);
+
+            if (currentUser is null)
+            {
+                return RedirectToAction("Index", "Home", new { });
+            }
```
Comment (maximsan) [inline DIMS-Core/DIMS-Core/Controllers/TaskTrackController.cs:156]: And you repeat the same logic in every method. Is it ok for you ? What do you think ?

**Review by maximsan — CHANGES_REQUESTED** [review]
Hm, I guess you have to work on improvement

File: DIMS-Core/DIMS-Core/Controllers/TaskTrackController.cs [hunk 7]
```diff
@@ -15,25 +16,37 @@ public class TaskTrackController : Controller
...
         }
 
         [HttpGet("")]
         public async Task<IActionResult> Index()
         {
-            // To Do - Get the id of the current user
-            int userId = 3;
+            if (!User.Identity.IsAuthenticated)
+            {
+                return RedirectToAction("Index", "Home", new { });
```
[the remark this answers is not in the document]
Comment (leonidbogin) [reply DIMS-Core/DIMS-Core/Controllers/TaskTrackController.cs]: https://github.com/Dev-incubator/DIMS-Core/pull/24/commits/e2ae2b625a627ecd32e0e39c38e38846b8a38df3

File: DIMS-Core/DIMS-Core/Controllers/TaskTrackController.cs (hunk 5 above)
Comment (leonidbogin) [reply DIMS-Core/DIMS-Core/Controllers/TaskTrackController.cs]: https://github.com/Dev-incubator/DIMS-Core/pull/24/commits/e2ae2b625a627ecd32e0e39c38e38846b8a38df3

File: DIMS-Core/DIMS-Core/Controllers/TaskTrackController.cs (hunk 6 above)
Comment (leonidbogin) [reply DIMS-Core/DIMS-Core/Controllers/TaskTrackController.cs]: https://github.com/Dev-incubator/DIMS-Core/pull/24/commits/e2ae2b625a627ecd32e0e39c38e38846b8a38df3

**Review by PushMeNow — APPROVED** [review]
Approve

**Pull request merged**